src/python/pants/engine/internals/scheduler_test.py

```diff
@@ -53,76 +53,11 @@ async def transitive_coroutine_rule(c: C) -> D:
     return D(b)
 
 
-@union
-class UnionBase:
-    pass
```

Comment:
The union tests were rewritten, but inspired by this. Mostly it's renamed to be less abstract.